In [3]:
from utils.loader import load_scene_graph
from data.robots import robots_withConfig
from planner.planner_initialPlanner2 import identify_defect_node, select_robot
       
user_description = "there is a crak on a wall in the room id 331"
scene_graph = load_scene_graph("/Users/nayunkim/Documents/GitHub/repairPlan/data/sceneGraphs/3dsg_withIds.json")
robot_DB = robots_withConfig

In [4]:
defect_id, room_id, reasoning = identify_defect_node(user_description, scene_graph)

defect_id=420 reasoning='The user mentioned a crack on a wall in room 331. Upon reviewing the 3D scene graph data, there is only one wall element associated with room 331, which is element id 420. Therefore, this is the most likely candidate for the defect node.' questions=None


In [5]:
print(defect_id, room_id, reasoning)

420 331 The user mentioned a crack on a wall in room 331. Upon reviewing the 3D scene graph data, there is only one wall element associated with room 331, which is element id 420. Therefore, this is the most likely candidate for the defect node.


420 None 1. The user's description specifies a crack on a wall with a window in room 331.
2. First, identify all wall elements in room 331 from the 3D scene graph. The elements with type 'Wall' and room '331' are:
   - Wall id 420
   - Wall id 806
   - Wall id 558
   - Wall id 643
3. Next, identify all window components in room 331. The components with type 'Window' and room '331' are:
   - Window id 139
   - Window id 266
   - Window id 267
4. Check the links to see which walls are connected to these windows:
   - Wall id 420 is linked to Window id 139
   - Wall id 806 is linked to Window id 266
   - Wall id 558 is linked to Window id 267
5. Since the user specified a crack on a wall with a window, we need to identify the wall that is linked to a window in room 331.
6. Wall id 420 is linked to Window id 139, which is in room 331.
7. Therefore, the defect node is Wall id 420, and it is located in room 331.

In [6]:
robot_id, reasoning_robot = select_robot(user_description, defect_id, robot_DB)

In [7]:
print(robot_id, reasoning_robot)

9 1. **Defect Analysis**: The defect is a crack on a wall located at coordinates (x: -0.26, y: -23.56855, z: 1.85) in room 331. The wall has a size of (x: 0.0, y: 11.78, z: 3.7). The crack is likely to require smoothing and possibly filling.

2. **Task Requirements**: The task involves smoothing the surface of the wall, which suggests the need for a robot capable of performing the 'smoothSurface' action. Additionally, the robot should be able to handle 'filler' material if filling is required.

3. **Robot Capability Analysis**:
   - **Robot 1**: Can perform 'smoothSurface' but has a max reach height of 2000 mm.
   - **Robot 2**: Can perform 'smoothSurface' with a max reach height of 3000 mm.
   - **Robot 3**: Can perform 'smoothSurface' with a max reach height of 3700 mm.
   - **Robot 4**: Can perform 'smoothSurface' but has a max reach height of 2000 mm.
   - **Robot 5**: Can perform 'smoothSurface' with a max reach height of 3700 mm.
   - **Robot 6**: Cannot perform 'smoothSurface'.
